augurpy/calculate_AUC.py

```diff
@@ -71,39 +75,110 @@ def draw_subsample(input: DataFrame, augur_mode: str, subsample_size: int, strat
             permuting the labels
         subsample_size: number of cells to subsample randomly per type from each experimental condition
         stratified: if `True` subsamples are stratified according to condition
+        random_state: set random seed
 
     Returns:
         Subsample of input of size subsample_size
     """
-    pass
+    if augur_mode == "permut":
+        # shuffle labels
+        y_columns = [col for col in adata.obs if col.startswith("y")]
+        adata.obs[y_columns] = adata.obs[y_columns].sample(frac=1, random_state=random_state).reset_index(drop=True)
```

Comment:
Why is it required to reset the index here?

Reply:
In permut mode we want to shuffle the cell labels, so that each cell will be assigned a random label. I f we don't reset the index, then then the labels will just be resorted and matched to the original cell later on because of matching indexes.